implementations/micrometer-registry-influx/src/shared/java/io/micrometer/influx/internal/LineProtocolBuilder.java

```diff
@@ -0,0 +1,185 @@
+ *
+ * @author Jakub Bednar (bednar@github) (13/05/2019 14:17)
+ */
+public class LineProtocolBuilder {
+
+    private final TimeUnit baseTimeUnit;
+    private final MeterRegistry.Config config;
+
+    public LineProtocolBuilder(final TimeUnit baseTimeUnit, final MeterRegistry.Config config) {
+        this.baseTimeUnit = baseTimeUnit;
+        this.config = config;
+    }
+
+    public Stream<String> writeMeter(Meter m) {
+        List<Field> fields = new ArrayList<>();
+        for (Measurement measurement : m.measure()) {
+            double value = measurement.getValue();
+            if (!Double.isFinite(value)) {
+                continue;
+            }
+            String fieldKey = measurement.getStatistic().getTagValueRepresentation()
+                    .replaceAll("(.)(\\p{Upper})", "$1_$2").toLowerCase();
+            fields.add(new Field(fieldKey, value));
+        }
+        if (fields.isEmpty()) {
+            return Stream.empty();
+        }
+        Meter.Id id = m.getId();
+        return Stream.of(influxLineProtocol(id, id.getType().name().toLowerCase(), fields.stream()));
+    }
+
+    public Stream<String> writeLongTaskTimer(LongTaskTimer timer) {
+        Stream<Field> fields = Stream.of(
+                new Field("active_tasks", timer.activeTasks()),
+                new Field("duration", timer.duration(baseTimeUnit))
+        );
+        return Stream.of(influxLineProtocol(timer.getId(), "long_task_timer", fields));
+    }
+
+    public Stream<String> writeCounter(Counter counter) {
+        Meter.Id id = counter.getId();
+        double count = counter.count();
+
+        return writeCounter(id, count);
+    }
+
+    public Stream<String> writeFunctionCounter(FunctionCounter counter) {
+        Meter.Id id = counter.getId();
+        double count = counter.count();
+
+        return writeCounter(id, count);
+    }
+
+    public Stream<String> writeGauge(Gauge gauge) {
+        Meter.Id id = gauge.getId();
+        double value = gauge.value();
+
+        return writeGauge(id, value);
+    }
+
+    public Stream<String> writeTimedGauge(TimeGauge timeGauge) {
+        Meter.Id id = timeGauge.getId();
+        double value = timeGauge.value(baseTimeUnit);
+
+        if (Double.isFinite(value)) {
+            return Stream.of(influxLineProtocol(id, "gauge", Stream.of(new Field("value", value))));
+        }
+        return Stream.empty();
+    }
+
+    public Stream<String> writeFunctionTimer(FunctionTimer timer) {
+        Stream<Field> fields = Stream.of(
+                new Field("sum", timer.totalTime(baseTimeUnit)),
+                new Field("count", timer.count()),
+                new Field("mean", timer.mean(baseTimeUnit))
+        );
+
+        return Stream.of(influxLineProtocol(timer.getId(), "histogram", fields));
+    }
+
+    public Stream<String> writeTimer(Timer timer) {
+        final Stream<Field> fields = Stream.of(
+                new Field("sum", timer.totalTime(baseTimeUnit)),
```

Comment:
What about percentiles of the histogram?

Reply:
see https://github.com/micrometer-metrics/micrometer/pull/1102

Reply:
I would like to follow the master branch. I will rebase #1423 after merge the #1102 into master.